### ISSUE TITLE
[Feature] Quality options for playlists

### ISSUE THREAD

**Issue by jamal-ali1989** [NickvisionApps/Parabolic#931]
When downloading a single video, one can get to choose the quality of the video to be downloaded. However, if the link is a playlist, this option is unavailable, even in "preferences".

**Comment by ekasprzak** [comment]
#892 seems related (however it shouldn't be about a video codec, eg. in my case I want to download playlist as audio).
From the info there, it should be fixed in v2024.10.2.

Raised by author for v2024.10.3, I can confirm it's still missing with current latest [v2024.11.1](https://github.com/NickvisionApps/Parabolic/releases/tag/2024.11.1)

**Comment by NoobGajen** [comment]
> #892 seems related (however it shouldn't be about a video codec, eg. in my case I want to download playlist as audio). From the info there, it should be fixed in v2024.10.2.
> 
> Raised by author for v2024.10.3, I can confirm it's still missing with current latest [v2024.11.1](https://github.com/NickvisionApps/Parabolic/releases/tag/2024.11.1)

Yeah, I installed `parabolic-gtk` from the AUR, and the `Quality` option for playlist downloads is not available.

**Comment by Mahantor** [comment]
same issue

**Comment by BOL1K** [comment]
Ok, thank you very much for the app and your support, good luck

**Comment by Mahantor** [comment]
It's been a long time since this request.
Are there any plans to add this feature?

**Comment by nlogozzo** [comment]
> Are there any plans to add this feature?

Yes. I just don't have as much free time anymore and this is a big feature to implement.

Currently when downloading playlists, Parabolic gets no information about the individual videos besides their name to keep it fast. The loading after validation would take 10x to scan each video for the qualities it supports. Trying to find a better way

**Comment by 65Monster65** [comment]
I think it’s been quite some time since this feature was discussed, hasn’t it? I’d really appreciate it if this could be implemented.

I often need to download playlists in 720p or 1080p, and having a toggle setting for “Flat Mode” or “Normal Mode” in the Parabolic preferences would make this much easier. Maybe this could be added as a simple switch in the settings, allowing users to choose whether to use --flat-playlist for quicker listing or the normal mode for full downloads.

It would also make the app feel more professional and user-friendly, especially for people who frequently download playlists.

Thanks for considering this feature! 🙏

**Comment by ghost** [comment]
I don't care if it is 10x slower to verify, Storage space is 100x more expensive to store, process,, and backup.

So even knowing the cost I'll take it.

**Comment by nlogozzo** [comment]
Okay upon further consideration, I've come up with my new UI and implementation for this.

We will no longer be using `--flat-playlist`. It severely hinders our ability to implement any of the following without it.

Inside the `Add Download` dialog, Parabolic will scan each individual media in a playlist to get its available formats, subtitles, time frames, etc.... (Note: this will take severely noticeable more time to validate a playlist than it currently does, *as we've mentioned above and in previous comments*). 

After validation, Parabolic will create a union of all the available media data and provide the following options in the dialog:
- General Options Page
    - Save Folder
        - Applied to every download in the playlist 
        - If the playlist is from a batch file, this will apply to downloads only that do not have a suggested save folder set from the batch file
    - File Type
        - Applied to every download in the playlist
    - Suggested Video Resolution
        - This will contain a combined list of all available video resolution across all of the formats of all of the media. Whichever resolution is suggested, will be the "golden" resolution that all downloads will try to download as. 
        - Remember, however, not every media will contain the same formats and resolution as others. Therefore, if a download is not available in this selected suggested resolution, the next lowest resolution will be used. (i.e. if the suggested resolution is 1080p all downloads will try to match that. If 1080p is not available for a download, 720p or lower will be used for that download instead).
        - Parabolic will never choose a higher resolution than the suggested unless no other lower resolution is available.
        - The `Preferred Video Codec` option in settings will be the only way to restrict these videos to download as a specific codec, otherwise it can be any
        - `Best` and `Worst` options will be available in the resolutions list as well to allow for downloading the best or worst resolution of any given media download respectively 
    - Suggested Audio Bitrate
        - This will contain a combined list of all available audio bitrates across all of the formats of all of the media. Whichever bitrate is suggested, will be the "golden" bitrate that all downloads will try to download as. 
        - Remember, however, not every media will contain the same formats and bitrates as others. Therefore, if a download is not available in this selected suggested bitrate, the next lowest bitrate will be used. (i.e. if the suggested bitrate is 129k all downloads will try to match that. If 129k is not available for a download, 118k or lower will be used for that download instead).
        - Parabolic will never choose a higher bitrate than the suggested unless no other lower bitrate is available.
        - The `Preferred Audio Codec` option in settings will be the only way to restrict these audios to download as a specific codec, otherwise it can be any
        - `Best` and `Worst` options will be available in the bitrates list as well to allow for downloading the best or worst bitrate of any given media download respectively 
- Items Page
    - This page will contain a list of all media available in a playlist for download 
    - Allow for Selection/Deselection of individual media for download
    - Allow for changing the file name of an individual media
    - Allow for changing the time frame of an individual media
    - Allow for numbering of titles of playlist items
    - Allow for reversing the download order of items
- Subtitles Page
    - This page will contain a combined list of all subtitles available across all media
    - The user can select multiple languages to be downloaded
        - If an individual media contains said subtitle language, it will be downloaded for it
        - If an individual media does not contain a selected subtitle language, that language will just be skipped for said individual media
- Advanced Options
    - Export M3U File for Playlist
    - Split Video by Chapters
        - Applied to every download in the playlist  
    - Export Descriptions
        - Applied to every download in the playlist  
    - Exclude from History
        - Applied to every download in the playlist  
    - Post Processor Argument
        - Applied to every download in the playlist

Please provide any comments, feedback, and/or suggestions on this new design before it becomes final. Thank you all for your continued patience on this feature and support of Parabolic! ❤️

**Comment by nlogozzo** [comment]
[V2026.2.0-beta1](https://github.com/NickvisionApps/Parabolic/releases/tag/2026.2.0-beta1-p1) is here with a fix for this. Notice it's a beta release, but I encourage you to test it out and use it as your main Parabolic version until stable.

Please let me know how it goes!

*If using flatpak, please allow some time for it to be available on flathub-beta*

**Comment by ghost** [comment]
2 bugs so far:
1. Everytime I set "include media id in title" and "include auto-generated subtitles" to off always resets those options to on. 
2. When I try to close out of parablic it has to be forced quit.

But other then those 2 bugs the looks, and interface have been hugely improved.❤️

**Comment by nlogozzo** [comment]
> 2 bugs so far:
> 1. Everytime I set "include media id in title" and "include auto-generated subtitles" to off always resets those options to on. 
> 2. When I try to close out of parablic it has to be forced quit.
> 
> But other then those 2 bugs the looks, and interface have been hugely improved.❤️

Are you running beta1 or stable? Make sure it's stable version. If you still see these issues please open new issue reports. Thanks

**Comment by ghost** [comment]
It's version 2026.2.0 so i think it's the stable version.

**Comment by nlogozzo** [comment]
Okay I will have this fixed in next version

**Comment by nlogozzo** [comment]
> When I try to close out of parablic it has to be forced quit.

Does it ask you to stop remaining downloads? What URLs did you try?

**Comment by ghost** [comment]
"Does it ask you to stop remaining downloads? What URLs did you try?"

1. no it does not, but after trying it the closing bug seemingly fixed it's self.

2. Normal youtube urls.

Which is weird because  it used to not close correctly even with nothing happening.

**Comment by nlogozzo** [comment]
I just released `2026.2.1`, wait for it to be available on Flathub and update and try again. If it still happens again, please open a new bug report issue.
